pkg/resource/patch.go

```diff
@@ -50,7 +51,11 @@ func (r *helper) Patch(name types.NamespacedName, kind, apiVersion string, patch
 }
 
 func (r *helper) setupPatchCommand(name, kind, apiVersion, patchType string, f cmdutil.Factory, patch string, ioStreams genericclioptions.IOStreams) (*kcmdpatch.PatchOptions, error) {
-
+	// This is bizarre and I don't know why it works, but it's the only way I could figure out
+	// to set the `manager` properly. HIVE-1744.
+	os.Args = []string{
```

Comment:
This one gave me much headache trying to find _some_ documentation around it, and I still don't like it - what if this implementation is very different on different os, and it causes bugs? Other places are harmless, but I really hope this one works as expected - atleast on unix systems.

Reply:
Seriously, this was trial and error. I get why the kubectl stuff we're importing would want to slurp from argv. What I can't figure out is why argv doesn't need to look like I expected. In particular, I initially tried
```go
os.Args = []string{
	"--field-manager", "hive8-"+string(r.controllerName),
}
```
...but ended up with `manager: --field-manager`!

I'm not worried about platform compatibility as this will always be running on a POSIX-compliant *ix.

Reply:
The most related [PR](https://github.com/golang/go/commit/96b79bd3208078835d6a1b9b8c7050554eccc734) I could find (which seriously isn't much) at least affirms that argv[1] cannot be empty when the eventual syscall happens, and when it is, they force it to be set as the parent function (the caller) before executing the child.
So I suppose we at least aren't causing a faux pas.